### DISCUSSION TITLE
Provide capability to configure route for durable functions monitor

### DISCUSSION THREAD

**Discussion by tomkerkhove** [scale-tone/DurableFunctionsMonitor#100] [Q&A] [answered]
Provide capability to configure route for durable functions monitor so that we can expose it on `{default route prefix}/monitor` where `monitor` is defined in the settings.

**Comment by scale-tone** [comment, marked as answer]
Today `monitor` is configurable via your CSPROJ-file with `DurableFunctionsMonitorRoutePrefix` setting (see [package readme](https://www.nuget.org/packages/DurableFunctionsMonitor.DotNetBackend)).

Configuring this via app settings would be great, but sadly it isn't possible - this value needs to be known at *compile* time (it is used by this [MSBUILD target](https://github.com/scale-tone/DurableFunctionsMonitor/blob/master/durablefunctionsmonitor.dotnetbackend/durablefunctionsmonitor.dotnetbackend.targets#L87) for "doctoring" function.json files after project build).

**Comment by scale-tone** [comment]
Is `DurableFunctionsMonitorRoutePrefix` setting what you were looking for, @tomkerkhove ?

**Comment by tomkerkhove** [comment]
It is still using the /api prefix then, but added this to the host.json:
```json
  "extensions": {
    "http": {
      "routePrefix": ""
    }
  },
```

However, when we add new HTTP functions we'll now always need to add the `/api` prefix to the route which is less ideal but I get why we have to do it.

_PS: feel free to enable GH Discussions for this kind of stuff :)_

**Comment by scale-tone** [comment]
> However, when we add new HTTP functions we'll now always need to add the /api prefix to the route which is less ideal but I get why we have to do it.

`DurableFunctionsMonitorRoutePrefix` is specifically designed to *avoid* doing that.
Once configured, DfMon should appear under `/{my-route-prefix}/{DurableFunctionsMonitorRoutePrefix}`.
If this is not what you're observing, then can you please post your CSPROJ file and your build output?

**Comment by tomkerkhove** [comment]
I didn't specify any prefix which is why `api` is added in front and the empty prefix was required

**Comment by scale-tone** [comment]
So @tomkerkhove , does the above resolve your issue? Or would you say something is still missing?

**Comment by tomkerkhove** [comment]
It's ok for now, thanks!
